packages/theme-check-common/src/checks/valid-local-blocks/index.spec.ts

```diff
@@ -0,0 +1,226 @@
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal('Local scoped blocks are not supported in theme blocks.');
+});
+
+it('should report errors when sections use theme blocks together with locally scoped blocks in root level', async () => {
+  const theme: MockTheme = {
+    'blocks/text.liquid': '',
+    'sections/local-blocks.liquid': `
+      {% schema %}
+      {
+        "name": "Section name",
+        "blocks": [
+          {
+            "type": "text"
+          },
+          {
+            "type": "local_block",
+            "name": "Local Block"
+          }
+        ]
+      }
+      {% endschema %}
+    `,
+  };
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal(
+    'Sections cannot use theme blocks together with locally scoped blocks.',
+  );
+});
+
+it('should report errors when sections use theme blocks together with locally scoped blocks in presets level', async () => {
+  const theme: MockTheme = {
+    'blocks/text.liquid': '',
+    'sections/local-blocks.liquid': `
+      {% schema %}
+      {
+        "name": "Section name",
+        "blocks": [
+          {
+            "type": "local_block",
+            "name": "Local Block"
+          }
+        ],
+        "presets": [
+          {
+            "name": "Default",
+            "blocks": [
+              {
+                "type": "text"
+              }
+            ]
+          }
+        ]
+      }
+      {% endschema %}
+    `,
+  };
+
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  console.log(offenses);
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal(
+    'Sections cannot use theme blocks together with locally scoped blocks.',
+  );
+});
+
+it('should report errors when sections use static theme blocks together with locally scoped blocks', async () => {
+  const theme: MockTheme = {
+    'sections/local-blocks.liquid': `
+          {% schema %}
+          {
+            "name": "Section name",
+            "blocks": [
+              {
+                "type": "local_block",
+                "name": "Local Block"
+              }
+            ],
+            "presets": [
+              {
+                "name": "Default",
+                "blocks": [
+                  {
+                    "id": "static_block",
+                    "type": "static_block",
+                    "static": true
+                  }
+                ]
+              }
+            ]
+          }
+          {% endschema %}
+        `,
+  };
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal(
+    'Sections cannot use static theme blocks together with locally scoped blocks.',
+  );
+});
+
+it('should report errors with correct indicies when sections use static theme blocks together with locally scoped blocks', async () => {
```

Comment:
```suggestion
it('should report errors with correct indices when sections use static theme blocks together with locally scoped blocks', async () => {
```

nit

Reply:
nit: maybe a bit overkill to have a test just for the indices, could maybe merge this with the previous test?